src/App.tsx

```diff
@@ -50,15 +44,18 @@
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector value={author} onChange={setAuthor} />
+                <UserSelector
+                  value={author}
+                  onChange={user => dispatch(setAuthor(user))}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
                 {!author && <p data-cy="NoSelectedUser">No user selected</p>}
 
-                {author && !loaded && <Loader />}
+                {author && !posts.loaded && <Loader />}
 
-                {author && loaded && hasError && (
+                {author && posts.loaded && (hasError || posts.hasError) && (
```

Comment:
The error handling logic checks both `hasError` and `posts.hasError`. Ensure that these represent the same error state or handle them separately if they are different. This will help avoid confusion and ensure that errors are handled appropriately.